iac-extensions/kubernetes/src/main/java/org/sonar/iac/kubernetes/visitors/AdjustableChecksVisitor.java

```diff
@@ -92,6 +103,31 @@ protected void reportIssue(@Nullable TextRange textRange, String message, List<S
       currentCtx.reportIssue(ruleKey, shiftedTextRange, message, enhancedAndAdjustedSecondaryLocations);
     }
 
+    private TextRange convertToHelmValuePathTextRange(TextRange shiftedTextRange) {
+      if (currentCtx instanceof HelmInputFileContext) {
+        var goTemplateTree = ((HelmInputFileContext) currentCtx).getGoTemplateTree();
+        var sourceWithComments = ((HelmInputFileContext) currentCtx).getSourceWithComments();
+        if (goTemplateTree != null && sourceWithComments != null) {
+          try {
+            var contents = currentCtx.inputFile.contents();
+            // The go template tree contains locations aligned to source code with additional trailing line numbers comments
+            var valuePathNodes = GoTemplateAstHelper.findValuePathNodes(goTemplateTree, shiftedTextRange, sourceWithComments);
+            var textRanges = valuePathNodes.map(FieldNode::location)
+              .map(location -> location.toTextRange(sourceWithComments))
+              .collect(Collectors.toList());
+            if (!textRanges.isEmpty()) {
+              // The text range may be too big, so it needs to be adjusted to the original source code
+              //TODO: When SONARIAC-1337 wil be implemented maybe this will be not needed anymore.
```

Comment:
Yes, I believe end goal of fixing the Go AST is to have node lengths in bounds of the input text